enabler/test/de/schildbach/pte/live/VrsProviderLiveTest.java

```diff
@@ -0,0 +1,497 @@
+		print(result);
+		assertEquals(QueryTripsResult.Status.OK, result.status);
+		assertTrue(result.trips.size() > 0);
+	}
+
+	@Test
+	public void testTripByCoord() throws Exception {
+		final QueryTripsResult result = queryTrips(new Location(LocationType.ANY, 50740530, 7129200), null, new Location(LocationType.ANY, 50933930, 6932440), new Date(), true, Product.ALL, WalkSpeed.NORMAL, Accessibility.NEUTRAL);
+		print(result);
+		assertEquals(QueryTripsResult.Status.OK, result.status);
+		assertTrue(result.trips.size() > 0);
+	}
+
+	@Test
+	public void testTripWithSurchargeInfo() throws Exception {
+		final QueryTripsResult result = queryTrips(new Location(LocationType.STATION, "687"), null, new Location(LocationType.STATION, "892"), new Date(), true, Product.ALL, WalkSpeed.NORMAL, Accessibility.NEUTRAL);
+		print(result);
+		assertEquals(QueryTripsResult.Status.OK, result.status);
+		assertTrue(result.trips.size() > 0);
+	}
+
+	private void manyRandomTrips(int latFrom, int latTo, int lonFrom, int lonTo) throws Exception {
+		Random rand = new Random(new Date().getTime());
+		int errors = 0;
+		long startTime = System.currentTimeMillis();
+		for (int i = 0; i < 5; i++) {
+			try {
+				int fromLat = latFrom + rand.nextInt(latTo - latFrom);
+				int fromLon = lonFrom + rand.nextInt(lonTo - lonFrom);
+				int toLat = latFrom + rand.nextInt(latTo - latFrom);
+				int toLon = lonFrom + rand.nextInt(lonTo - lonFrom);
+				final QueryTripsResult result = queryTrips(new Location(LocationType.ANY, fromLat, fromLon), null,
+						new Location(LocationType.ANY, toLat, toLon), new Date(), true, Product.ALL, WalkSpeed.NORMAL,
+						Accessibility.NEUTRAL);
+				System.out.println("# " + (i+1));
+				if (result.status.equals(QueryTripsResult.Status.OK)) {
+					print(result);
+				} else {
+					System.out.println("Status is " + result.status);
+					errors++;
+				}
+			} catch (SocketTimeoutException ex) {
+				System.out.println("SocketTimeoutException: " + ex);
+				errors++;
+			}
+		}
+		final long stopTime = System.currentTimeMillis();
+		final long elapsedTime = stopTime - startTime;
+		System.out.println("Elapsed: " + (elapsedTime / 1000) + " seconds");
+		System.out.println("Errors: " + errors);
+	}
+
+	@Ignore
+	@Test
+	public void manyRandomTripsNRW() throws Exception {
+		manyRandomTrips(50500000, 51600000, 6200000, 7600000);
+	}
+
+	@Ignore
+	@Test
+	public void manyRandomTripsCologne() throws Exception {
+		manyRandomTrips(50828176, 51083369, 6770942, 7161643);
+	}
+
+	@Ignore
+	@Test
+	public void manyRandomTripsBonn() throws Exception {
+		manyRandomTrips(50632639, 50774408, 7019582, 7209096);
+	}
+
+	@Ignore
+	@Test
+	public void manyRandomTripsDuesseldorf() throws Exception {
+		manyRandomTrips(51123960, 51353094, 6689381, 6940006);
+	}
+
+	private static class LocationComparator implements Comparator<Location> {
+		public int compare(Location o1, Location o2) {
+			return o1.name.compareTo(o2.name);
```

Comment:
I suggest using "ComparisonChain.start().compare(o1.name, o2.name).result()" – Location.name can be null.